libsel4test/include/sel4test/test.h

```diff
@@ -96,16 +96,10 @@ typedef struct test_type {
     test_result_t (*run_test)(struct testcase *test, uintptr_t e);
 } ALIGN(32) test_type_t;
 
-#if defined(__has_attribute) && __has_attribute(retain)
-#define ATTR_USED_RETAIN __attribute__((used,retain))
-#else
-#define ATTR_USED_RETAIN __attribute__((used))
-#endif
-
```

Comment:
We still want to keep this check though, so instead of a Cmake check, can't we add `defined(__clang__)` here?